### PR TITLE
Examples: Adding more tags to search.

### LINKED ISSUE #18700: Examples: add tags support to search
e.g. rename webgl_postprocessing_unreal_bloom_selective.html to webgl_postprocessing_glow_unreal_bloom_selective.html, and the other one too.

because people typing "glow" in examples search box get nothing.

### PR DESCRIPTION
**Pull request by munrocket**
Fixes #18700
Related commits: #18773, #18779

### CODE REVIEW

**Comment by Mugen87** [comment]
Does this PR close #18779? (I don't think we need two of them)

**Comment by makc** [comment]
> file.concat( tags[ file ] )

interesting. but dont you think you need to put a separator there

File: examples/files.js [hunk 1]
```diff
@@ -391,3 +391,107 @@ var files = {
...
+  "webgl_postprocessing_unreal_bloom": "glow",
+  "webgl_postprocessing_unreal_bloom_selective": "glow",
+  "webgl_fire": "smoke",
+  "webgl_materials_modified": "onBeforeCompile",
+  "webgl_shadowmap_csm": "cascaded",
+  "webgl_shadowmap_pcss": "soft",
+  "webgl_simple_gi": "global illumination",
+  "webgl_tiled_forward": "derivatives",
+  "webgl2_multisampled_renderbuffers": "msaa",
+	"physics_ammo_cloth": "integration",
```
Comment (Mugen87) [inline examples/files.js:490]: Minor: It seems there are some small formatting issues...

**Comment by munrocket** [comment]
>file.concat( tags[ file ] ).match( exp );

@makc yep, but probably we don't need it, since we using simple substring matching.
We should change it to `.includes("word")`.

**Comment by makc** [comment]
@munrocket 
> since we using simple substring matching

and that is exactly why you need it, if one of your words ends in `bit` and another starts with `ch` you get unwanted matches

**Comment by munrocket** [comment]
[githack](https://rawcdn.githack.com/munrocket/three.js/5eedf013c972e52bfd5c3af46b595cdea13d052b/examples/index.html?q=wat)

**Comment by munrocket** [comment]
[githack](https://rawcdn.githack.com/munrocket/three.js/c187376dcb2d94b5f8f335aaf9d9025061c167af/examples/index.html), found an error.

**Comment by mrdoob** [comment]
> adding a new section "searched by tags" in UI

I find that confusing... I think I prefer #18779's simplicity 🤔

**Comment by munrocket** [comment]
In this case search results appearing silently without reason.
We can change a color in “searched by tag” section and show tags to each example.

**Comment by Mugen87** [comment]
> We can change a color in “searched by tag” section and show tags to each example.

The search is intended to filter the example list, not adding new items to it. This is what makes your PR confusing.

**Comment by mrdoob** [comment]
Yes, I think #18779 was fine, just had not had a chance to review it properly.
But this PR has more tags.. 😅

**Comment by Mugen87** [comment]
> But this PR has more tags

How about merging #18779 first. @munrocket can then rebase this PR and add the tags.

**Comment by makc** [comment]
well technically munrocket did already include 18779 in this PR. it is just that his next commits overwrote the code then )

**Comment by munrocket** [comment]
Yep, I am force pushed head to 5eedf01. Technically d5a8cb5 increasing commit from @makc

**Comment by mrdoob** [comment]
@munrocket Yes, but... unfortunately, the commit where you added more tags also has the UI code that it's not ready to merge yet. https://github.com/mrdoob/three.js/pull/19245/commits/5eedf013c972e52bfd5c3af46b595cdea13d052b

**Comment by makc** [comment]
@mrdoob since #18700 is now closed, where do we track the missing/needed tags? here?

I just went to https://raw.githack.com/mrdoob/three.js/dev/examples/#webgl_animation_cloth and I could not find map controls by searching for `drag` or `pan` so this girl is still out of luck:

**Comment by Mugen87** [comment]
> @mrdoob since #18700 is now closed, where do we track the missing/needed tags? here?

Yes. @munrocket just needs to rebase the PR and remove the UI related code.

File: examples/index.html [hunk 2]
```diff
@@ -249,7 +249,7 @@ <h1><a href="https://threejs.org">three.js</a></h1>
 
 			var link = links[ file ];
 			var name = getName( file );
-			if ( file in tags ) file += tags[ file ].join( ' ' );
+			if ( file in tags ) file += tags[ file ];
```
Comment (Mugen87) [inline examples/index.html:252]: This change should not be necessary.

Comment (munrocket) [reply examples/index.html:252]: I just dumped a old version. I am a lazy cookie.

Comment (munrocket) [reply examples/index.html:252]: By the way, I am also added webgl_loader_ttf with "text, font"

**Comment by mrdoob** [comment]
> @mrdoob since #18700 is now closed, where do we track the missing/needed tags? here?

Just reopened the #18700.

File: examples/index.html (hunk 2 above)
Comment (Mugen87) [reply examples/index.html:252]: It seems this change still needs to be undone, no?

Comment (mrdoob) [reply examples/index.html:252]: With this approach I think if you search for `,` you'll get a lot of results?
I think I prefer the array approach.

Comment (munrocket) [reply examples/index.html:252]: Doesn’t see why, sorry. Splitting into tags and then join it again in code?
Looks like cargo cult to Tags :D . The same thing with ` `  will happens after .join().
Real problem not here. In UX probably. Nobody searching `,`.

Comment (munrocket) [reply examples/index.html]: Fixed it. You forcing me to make a drama out of it.

**Review by Mugen87 — APPROVED** [review]

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**